website/source/guides/operations/vault-ha-consul.html.md

```diff
@@ -0,0 +1,621 @@
+---
+layout: "guides"
+page_title: "Vault HA with Consul - Guides"
+sidebar_current: "guides-operations-vault-ha"
+description: |-
+  This guide This guide will walk you through a simple Vault Highly Available
+  (HA) cluster implementation. While this is not an exhaustive or prescriptive
+  guide that can be used as a drop-in production example, it covers the basics
+  enough to inform your own production setup.
+---
+
+# Vault Haigh Availability (HA)
+
+Vault can run in a high availability (HA) mode to protest against  outages by
+running multiple Vault servers. Vault is typically bound by the IO limits of the
+storage backend rather than the compute requirements. Certain storage backends,
+such as Consul, provide additional coordination functions that enable Vault to
+run in an HA configuration while others provide a more robust backup and
+restoration process.
+
+This guide will walk you through a simple Vault Highly Available (HA) cluster
+implementation. While this is not an exhaustive or prescriptive guide that can
+be used as a drop-in production example, it covers the basics enough to inform
+your own production setup.
+
+
+## Reference Materials
+
+- [High Availability Mode](/docs/concepts/ha.html)
+- [Consul Storage Backend](/docs/configuration/storage/consul.html)
+- [High Availability Parameters](/docs/configuration/index.html#high-availability-parameters)
+- [Consul Agent Configuration](https://www.consul.io/docs/agent/options.html)
+
+## Estimated Time to Complete
+
+10 minutes
+
+## Prerequisites
+
+- Vault
+- Consul
+
+## Steps
+
+Our goal in following this guide is to arrive at a Vault HA setup
+consisting of the following:
+
+- 2 Vault servers: 1 active and 1 standby
+- Cluster of 3 Consul servers
+
+### Reference Diagram
+
+This diagram lays out the simple architecture details for reference:
+
+![Reference Architecture](/assets/images/vault-ha-consul.png)
+
+We’ll use the open source software editions of Vault and Consul in this guide,
+but the setup is the same for Enterprise editions.
+
+### Consul Server Cluster
+
+Our Consul servers in this guide will be defined by IP address only, but also
+referenced by a label:
+
+- **`consul_s1: 10.1.42.101`**
+- **`consul_s2: 10.1.42.102`**
+- **`consul_s3: 10.1.42.103`**
+
+The consul binary is presumed to be located in /usr/local/bin/consul, but if
+your case differs, you can adjust the path references accordingly.
+
+With that in mind, here is a basic Consul server configuration starting point:
+
+#### Consul Server Configuration
+
+```plaintext
+{
+  "server": true,
+  "node_id": "$NODE_ID",
```

Comment:
This will need updating in the Consul configuration throughout as we'd later learned that it's `node_name` / `$NODE_NAME` we want here instead so this line should read:

```
"node_name": "$NODE_NAME",
```

Reply:
Fixed. Thx!